### PR TITLE
feat: Addition of support for new Gcp pub sub audit log integration

### PR DESCRIPTION
**Pull request by djmctavish**
***Issue***: Include link to the Jira/Github Issue
https://lacework.atlassian.net/browse/GROW-1399

***Description:***
Addition of new GCP Pub Sub Audit Log integration 

***Additional Info:***
Include any other relevant information such as how to use the new functionality, screenshots, etc.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.18
 require (
 	github.com/gruntwork-io/terratest v0.41.3
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.24.1
-	github.com/lacework/go-sdk v1.1.3-0.20221214173411-1ec57f679902
+	github.com/lacework/go-sdk v1.9.2-0.20230207104645-f46a8c8ec140
```
Comment (dmurray-lacework) [inline go.mod:8]: should we update go-sdk dep to latest `v1.12.0`

File: integration/resource_lacework_integration_gcp_al_pub_sub_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+			EnvVars:      tokenEnvVar,
+			Vars: map[string]interface{}{
+				"name":             "GCP pub sub audit log integration example",
+				"client_id":        gcreds.ClientID,
+				"client_email":     gcreds.ClientEmail,
+				"private_key_id":   gcreds.PrivateKeyID,
+				"private_key":      gcreds.PrivateKey,
+				"integration_type": "PROJECT",
+				"project_id":       gcreds.ProjectID,
+				"subscription":     "projects/techally-test/subscriptions/lw-at-techally-test-lacework-subscription-44c23c7d",
```
Comment (dmurray-lacework) [inline integration/resource_lacework_integration_gcp_al_pub_sub_test.go:29]: Can we make this an env var too

File: website/docs/r/integration_gcp_pub_sub_audit_log.html.markdown [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	}
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The GCP Audit Trail integration name.
+* `organization_id` - (Optional) The organization ID. Required if `integration_type` is set to `ORGANIZATION`.
```
Comment (dmurray-lacework) [inline website/docs/r/integration_gcp_pub_sub_audit_log.html.markdown:36]: nit: I think we sort required params first. Ignore if this is not the case on other resources markdown

**Review by dmurray-lacework — APPROVED** [review]

File: examples/resource_lacework_integration_gcp_al_pub_sub/main.tf [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  name = var.name
+  credentials {
+    client_id      = var.client_id
+    client_email   = var.client_email
+    private_key_id = var.private_key_id
+    private_key    = var.private_key
+  }
+  integration_type = var.integration_type
+  project_id       = var.project_id
+  subscription     = var.subscription
```
Comment (afiune) [inline examples/resource_lacework_integration_gcp_al_pub_sub/main.tf:64]:
```suggestion
  subscription_id     = var.subscription
```

File: examples/resource_lacework_integration_gcp_al_pub_sub/main.tf [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  type    = string
+  default = "PROJECT"
+}
+
+variable "project_id" {
+  type    = string
+  default = "example-project-id"
+}
+
+variable "subscription" {
```
Comment (afiune) [inline examples/resource_lacework_integration_gcp_al_pub_sub/main.tf:44]:
```suggestion
variable "subscription_id" {
```

File: website/docs/r/integration_gcp_pub_sub_audit_log.html.markdown (hunk 3 above)
Comment (djmctavish) [reply website/docs/r/integration_gcp_pub_sub_audit_log.html.markdown:36]: Seems to be a mixture

File: integration/resource_lacework_integration_gcp_al_pub_sub_test.go (hunk 2 above)
Comment (djmctavish) [reply integration/resource_lacework_integration_gcp_al_pub_sub_test.go]: Now uses` LW_PUB_SUB_SUBSCRIPTION` env

File: go.mod (hunk 1 above)
Comment (djmctavish) [reply go.mod]: Updated to use the latest SDK

File: examples/resource_lacework_integration_gcp_al_pub_sub/main.tf (hunk 5 above)
Comment (djmctavish) [reply examples/resource_lacework_integration_gcp_al_pub_sub/main.tf]: all references to `subscription` have been updated to `subscription_id` throughout the code

**Pull request merged**